.github/workflows/pull-request.yaml

```diff
@@ -50,13 +50,14 @@ jobs:
       uses: actions/setup-go@v5
       with:
         go-version: "1.21.0"
-    - name: Build Linux AMD64
+    - name: Build Linux AMD64 and Output Metadata
       run: |
         export SHORT_SHA=dev-$(git rev-parse --short=7 HEAD)
-        make embedded-cluster-linux-amd64 VERSION="${SHORT_SHA}"
+        make -B embedded-cluster-linux-amd64 K0S_VERSION=$(make print-PREVIOUS_K0S_VERSION) VERSION="${SHORT_SHA}-previous-k0s"
+        tar -C output/bin -czvf embedded-cluster-linux-amd64-previous-k0s.tgz embedded-cluster
+        ./output/bin/embedded-cluster version metadata > metadata-previous-k0s.json
+        make -B embedded-cluster-linux-amd64 VERSION="${SHORT_SHA}"
```

Comment:
running make with `-B` to make sure it re-builds everything for the old/new versions (goods and go modules)